### PR TITLE
Add updatedAt for ilm-rule-ls

### PR DESCRIPTION
**Pull request by krisis**
## Description
This will allow users to learn the most recent time when their bucket's ILM policy was updated.
Depends on https://github.com/minio/minio-go/pull/1831
## Motivation and Context
Bucket ILM policies may be updated multiple times depending on an organization's evolving object lifecycle requirements. It can become difficult to reason with object lifecycle events like expiration, transition, etc without knowing when the ILM policy was last updated.

## How to test this PR?
1. Configure ILM policy for a bucket
2. `mc ilm ls --json ALIAS/bucket`
<details open> <summary>Sample output</summary>

```json
{
 "status": "success",
 "target": "myminio/bucket",
 "config": {
  "Rules": [
   {
    "Expiration": {
     "Days": 200
    },
    "ID": "chn7t9lvvqdakbhfn73g",
    "Status": "Enabled"
   },
   {
    "ID": "chn934gd6vv3cmvlc5s0",
    "NoncurrentVersionExpiration": {
     "NoncurrentDays": 1
    },
    "Status": "Enabled"
   }
  ]
 },
 "updatedAt": "2023-05-24T22:37:06Z"
}
```
</details>

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Review by poornas — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -120,9 +120,11 @@ require (
...
 	golang.org/x/sync v0.1.0 // indirect
-	golang.org/x/sys v0.7.0
+	golang.org/x/sys v0.8.0
 	google.golang.org/genproto v0.0.0-20230403163135-c38d8f061ccd // indirect
 	google.golang.org/grpc v1.54.0 // indirect
 	google.golang.org/protobuf v1.30.0 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
 )
+
+replace github.com/minio/minio-go/v7 v7.0.52 => github.com/krisis/minio-go/v7 v7.0.0-20230524223141-ad61f9f521a9
```
Comment (shtripat) [inline go.mod:130]: Will this be reverted once base PR of minio-go merged? I understand so.

**Review by shtripat — APPROVED** [review]
LGTM. Verified with `minio` [PR#17271](https://github.com/minio/minio/pull/17271/files)

File: go.mod (hunk 1 above)
Comment (krisis) [reply go.mod:130]: Yes, I will update this PR to use github.com/minio/minio-go/v7.

Comment (krisis) [reply go.mod]: Done

**Pull request merged**